### DISCUSSION TITLE
Support for real-time collaboration in the extension?

### DISCUSSION THREAD

**Discussion by grisaitis** [microsoft/vscode-jupyter#12933] [Questions and Answers]
Let's say I have:
1. on a remote host: a jupyter lab server running with the `--collaborative` flag ([jupyter lab docs](https://jupyterlab.readthedocs.io/en/stable/user/rtc.html))
2. on my local machine: VSCode, connected via tunnel or SSH to the remote host, with the Jupyter extension
3. a notebook file open in VSCode and in the jupyter web app
4. in VSCode, I've provided the URL of the remote jupyter lab server (so I'm not running a jupyter server locally, separately) 

My question: can I have a notebook open on the remote host _and_ in VSCode, and collaboratively edit the notebook in VSCode? That is, when I edit the notebook in VSCode, I'd like to see the changes in real-time on the jupyter web client as well. 

Asking because VSCode is amazing for editing, but the jupyter web client is better (more responsive) for executing the cells. I'd like the best of both worlds 😃 

In theory I think this might (should?) be possible. RTC works with different web clients, so maybe it would work with a VSCode client as well?

Thanks!

**Comment by DonJayamanne** [comment]
Thank you for posting this question
Unfortunately this is not something that is currently supported 

vscode edits ipynb files as plain json files even without Jupyter application installed
Vscode supports collaboration within vscode itself via live share
